### PR TITLE
[PowerPC] Expand FSINCOS of fp128

### LINKED ISSUE #76442: [PowerPC] fatal error: error in backend: Cannot select: 0x8e697e8320: f128,f128 = fsincos 0x8e697eb450
Clang 17.0.6 on Linux (Gentoo ppc64le) and FreeBSD 15.0 (powerpc64le)

Test case:
```
# 1 "" 3
cpowl() { int theta = cosl(theta) * sinl(theta); }
```

Compile with:
`/usr/lib/llvm/17/bin/clang-17 -cc1 -emit-obj -target-cpu ppc64le -mabi=ieeelongdouble -Werror -Weverything -Wno-gnu-line-marker -fgnuc-version=4.2.1 s_cpowl-062d88.c`

Output:
```
fatal error: error in backend: Cannot select: 0x6d420285c0: f128,f128 = fsincos 0x6d4202b6f0
  0x6d4202b6f0: f128,ch,glue = CopyFromReg 0x6d4202b680, Register:f128 $v2, 0x6d4202b680:1
    0x6d42028b70: f128 = Register $v2
    0x6d4202b680: ch,glue = callseq_end 0x6d4202b610, TargetConstant:i64<32>, TargetConstant:i64<0>, 0x6d4202b610:1
      0x6d420288d0: i64 = TargetConstant<32>
      0x6d42028940: i64 = TargetConstant<0>
      0x6d4202b610: ch,glue = PPCISD::CALL_NOP 0x6d4202b530, TargetExternalSymbol:i64'__floatsikf', Register:i64 $x3, Register:i64 $x2, RegisterMask:Untyped, 0x6d4202b530:1
        0x6d4202b5a0: i64 = TargetExternalSymbol'__floatsikf'
        0x6d42028e80: i64 = Register $x3
        0x6d42028cc0: i64 = Register $x2
        0x6d42028d30: Untyped = RegisterMask
        0x6d4202b530: ch,glue = CopyToReg 0x6d420284e0, Register:i64 $x3, 0x6d42028320
          0x6d42028e80: i64 = Register $x3
          0x6d42028320: i64,ch = load<(dereferenceable load (s32) from %ir.theta), sext from i32> 0x6d41fd2050, FrameIndex:i64<1>, undef:i64
            0x6d420282b0: i64 = FrameIndex<1>
            0x6d42028390: i64 = undef
In function: cpowl
```

This issue has been confirmed to happen on Linux, but its origin is WIP patch to switch FreeBSD/powerpc64le to IEEE long double. The issue happens when building msun library (FreeBSD's libm).

### PR DESCRIPTION
**Pull request by ecnelises**
Fixes #76442

### CODE REVIEW

File: llvm/test/CodeGen/PowerPC/f128-arith.ll [hunk 1]
```diff
@@ -419,6 +419,66 @@ entry:
 }
 declare fp128 @llvm.cos.f128(fp128 %Val)
 
+define fp128 @qp_sincos(ptr nocapture readonly %a) {
+; CHECK-LABEL: qp_sincos:
+; CHECK:       # %bb.0: # %entry
+; CHECK-NEXT:    mflr r0
+; CHECK-NEXT:    stdu r1, -64(r1)
+; CHECK-NEXT:    std r0, 80(r1)
+; CHECK-NEXT:    .cfi_def_cfa_offset 64
```
Comment (chenzheng1030) [inline llvm/test/CodeGen/PowerPC/f128-arith.ll:428]: nit: remove these unnecessary cfi pseudo instructions.

File: llvm/lib/Target/PowerPC/PPCISelLowering.cpp [hunk 2]
```diff
@@ -1176,6 +1176,7 @@ PPCTargetLowering::PPCTargetLowering(const PPCTargetMachine &TM,
       setTruncStoreAction(MVT::f128, MVT::f32, Expand);
 
       // No implementation for these ops for PowerPC.
+      setOperationAction(ISD::FSINCOS, MVT::f128, Expand);
```
Comment (chenzheng1030) [inline llvm/lib/Target/PowerPC/PPCISelLowering.cpp:1179]: Have you checked other math related ISD nodes? Are MVT::f128 correctly handled for them?

Comment (ecnelises) [reply llvm/lib/Target/PowerPC/PPCISelLowering.cpp:1179]: Yes, according to https://en.cppreference.com/w/c/numeric/math and nodes defined in ISDOpcodes.h. (they were already tested in f128-arith/round/fma.ll)

**Review by bzEq — APPROVED** [review]
LGTM.

**Review by chenzheng1030 — APPROVED** [review]
LGTM

**Pull request merged**